eql/utils.py

```diff
@@ -243,6 +243,30 @@ def is_stateful(query):
     return any(isinstance(node, stateful_nodes) for node in query)
 
 
+def get_query_type(query):
+    """Get the type of a query (sequence/join/event).
+
+    :param ast.PipedQuery|ast.Analytic query: The parsed query AST to analyze
+    :rtype: str
+    """
+    from . import ast  # noqa: E402
+
+    if isinstance(query, ast.EqlAnalytic):
+        query = query.query
+
+    elif not isinstance(query, ast.PipedQuery):
+        raise TypeError("unsupported type {} to eql.utils.get_query_type. Expected {}".format(type(query), ast.EqlNode))
```

Comment:
Can you create a custom exception for this which inherits the base eql exception and `TypeError` to catch this easier

Reply:
I think it's okay in these utility functions. It'd be the same exception as if you called with one arg or too many args. It's more of a curse of dynamic typing than anything else. So it's not an EQL-specific error, but a "you used python wrong" error

Reply:
I think in `match_kv` that should probably be an EqlError instead

Reply:
Or rather, use `parse_field` instead